calus.js

```diff
@@ -1,228 +1,232 @@
+                    let isAvailable = false;
 
-
-                        let startOf = day.startOf('day')
-
-                        days.push({
-                            time: day,
-                            isToday: day.hasSame(this.now, 'day'),
-                            isAvailable: isAvailable,
-                            isPast: startOf < startOfToday,
-                            isFuture: startOf > startOfToday,
-                            isThisMonth: day.hasSame(date, 'month'),
-                        })
+                    if (available.length && available[0].day === day.day) {
+                        isAvailable = true;
                     }
 
-                    months.push({
-                        time: date.startOf('month'),
-                        isCurrentMonth: +days[10].time.startOf('month') == +startOfToday.startOf('month'),
-                        isInCurrentYear: +days[10].time.startOf('year') == +startOfToday.startOf('year'),
-                        days: days
-                    })
+                    let startOf = day.startOf("day");
 
-                    date = date.plus({ months: 1 })
+                    days.push({
+                        time: day,
+                        isToday: day.hasSame(this.now, "day"),
+                        isAvailable: isAvailable,
+                        isPast: startOf < startOfToday,
+                        isFuture: startOf > startOfToday,
+                        isThisMonth: day.hasSame(date, "month"),
+                    });
                 }
 
-                return months
+                months.push({
+                    time: date.startOf("month"),
+                    isCurrentMonth:
+                        +days[10].time.startOf("month") ==
+                        +startOfToday.startOf("month"),
+                    isInCurrentYear:
+                        +days[10].time.startOf("year") ==
+                        +startOfToday.startOf("year"),
+                    days: days,
+                });
+
+                date = date.plus({ months: 1 });
             }
-        },
-        methods: {
-            // select a day
-            select: function (day) {
-                if (day.isAvailable) {
-                    // selected state is handled by direct DOM control
-                    // it's a bit hacky, but it prevents a full redraw
-                    // of the calendar, making it feel a lot snappier,
-                    // especially with large amount of displayed dates
-                    this.resetSelected()
-                    this.selected = day.time
 
-                    this.addSelectedStyle(this.selected)
+            /**
+             * Make sure everything is set correctly
+			 * before disabling scroll
+             */
+            if (
+                this.monthControlsClasses.length &&
+                !this.displayInColumn &&
+                !this.allowScrollingOutsideOfDateRange
+            ) {
```

Comment:
Be better if these checks were in the `disableScroll()` function
It could be called from the template remember
Would suggest if it won't run then do a `console.warn()`

Reply:
done